pkg/tasks/prometheus.go

```diff
@@ -125,9 +125,9 @@ func (t *PrometheusTask) Run(ctx context.Context) error {
 		return errors.Wrap(err, "initializing Prometheus RBAC proxy Secret failed")
 	}
 
-	err = t.client.CreateIfNotExistSecret(ctx, rs)
+	err = t.client.CreateOrUpdateSecret(ctx, rs)
```

Comment:
why do we need to update the secret here?

Reply:
Because the existing secret already contains configuration for authorization through token access reviews. Now we need to update the config to change the authz method.

Reply:
thanks,  i may need make a similar change to my PR, too 🤔
https://github.com/openshift/cluster-monitoring-operator/pull/1392